cpp/RNSScreenRemovalListener.cpp

```diff
@@ -0,0 +1,25 @@
+#include "RNSScreenRemovalListener.h"
+#include <react/renderer/mounting/ShadowViewMutation.h>
+using namespace facebook::react;
+
+std::optional<MountingTransaction> RNSScreenRemovalListener::pullTransaction(
+    SurfaceId surfaceId,
+    MountingTransaction::Number transactionNumber,
+    const TransactionTelemetry &telemetry,
+    ShadowViewMutationList mutations) const {
+  for (ShadowViewMutation mutation : mutations) {
+    if (mutation.type == ShadowViewMutation::Type::Remove &&
+        mutation.oldChildShadowView.componentName != nullptr &&
```

Comment:
Why do we check `oldChildShadowView.componentName` for nullish value while later using `parentShadowView.componentName` for comparison purposes?

Also, are we looking for `RNSScreenStack` here, cause we want this fix to work only for native-stack? Isn't this (whiteflash) also an issue with ScreenContainer? (I don't have knowledge here, just wondering reading through the code).

Reply:
> Why do we check oldChildShadowView.componentName for nullish value while later using parentShadowView.componentName for comparison purposes?

Because we use `oldChildShadowView.componentName` in `listenerFunction_`.

> Also, are we looking for RNSScreenStack here, cause we want this fix to work only for native-stack? Isn't this (whiteflash) also an issue with ScreenContainer? (I don't have knowledge here, just wondering reading through the code).

Indeed it is only needed in `RNSScreenStack` since other navigators do not remove the view until the animation is completed.